app/src/main/java/com/bunbeauty/papakarlo/feature/cafe/screen/cafe_list/CafeListFragment.kt

```diff
@@ -8,62 +8,107 @@ import androidx.compose.foundation.layout.padding
 import androidx.compose.foundation.lazy.LazyColumn
 import androidx.compose.foundation.lazy.itemsIndexed
 import androidx.compose.runtime.Composable
-import androidx.compose.runtime.collectAsState
+import androidx.compose.runtime.LaunchedEffect
 import androidx.compose.runtime.getValue
 import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.stringResource
 import androidx.compose.ui.tooling.preview.Preview
+import androidx.lifecycle.compose.ExperimentalLifecycleComposeApi
+import androidx.lifecycle.compose.collectAsStateWithLifecycle
+import androidx.navigation.fragment.findNavController
 import by.kirich1409.viewbindingdelegate.viewBinding
 import com.bunbeauty.papakarlo.R
 import com.bunbeauty.papakarlo.common.BaseFragment
-import com.bunbeauty.papakarlo.common.state.State
 import com.bunbeauty.papakarlo.common.ui.screen.ErrorScreen
 import com.bunbeauty.papakarlo.common.ui.screen.LoadingScreen
 import com.bunbeauty.papakarlo.common.ui.theme.FoodDeliveryTheme
+import com.bunbeauty.papakarlo.common.ui.toolbar.FoodDeliveryCartAction
+import com.bunbeauty.papakarlo.common.ui.toolbar.FoodDeliveryToolbarScreen
 import com.bunbeauty.papakarlo.databinding.FragmentCafeListBinding
 import com.bunbeauty.papakarlo.extensions.setContentWithTheme
 import com.bunbeauty.papakarlo.feature.cafe.model.CafeItem
 import com.bunbeauty.papakarlo.feature.cafe.ui.CafeItem
+import com.bunbeauty.papakarlo.feature.product_details.ProductDetailsFragmentDirections
+import com.bunbeauty.papakarlo.feature.top_cart.TopCartUi
 import com.bunbeauty.shared.domain.model.cafe.CafeStatus
 import com.google.android.material.transition.MaterialFadeThrough
+import org.koin.android.ext.android.inject
 import org.koin.androidx.viewmodel.ext.android.viewModel
 
 class CafeListFragment : BaseFragment(R.layout.fragment_cafe_list) {
 
     override val viewBinding by viewBinding(FragmentCafeListBinding::bind)
     override val viewModel: CafeListViewModel by viewModel()
+    private val cafeListUiStateMapper: CafeListUiStateMapper by inject()
 
     override fun onCreate(savedInstanceState: Bundle?) {
         super.onCreate(savedInstanceState)
         enterTransition = MaterialFadeThrough()
     }
 
+    @OptIn(ExperimentalLifecycleComposeApi::class)
     override fun onViewCreated(view: View, savedInstanceState: Bundle?) {
         overrideBackPressedCallback()
         super.onViewCreated(view, savedInstanceState)
 
         viewModel.getCafeItemList()
         viewBinding.fragmentCafeListCvMain.setContentWithTheme {
-            val cafeItemList by viewModel.cafeItemList.collectAsState()
-            CafeListScreen(cafeItemList)
+            val cafeItemList by viewModel.cafeListState.collectAsStateWithLifecycle()
+            CafeListScreen(
+                cafeListState = cafeListUiStateMapper.map(cafeItemList),
+                onCafeClicked = viewModel::onCafeCardClicked,
```

Comment:
We don't need to pass every lambda as a parameter. Calls to `viewModel` don't affect `@Previews`